### ISSUE TITLE
Add option to delete user from the user API

### ISSUE THREAD

**Issue by eldadfux** [appwrite/appwrite#378]
We should allow to delete users from both the users API and the console. One important thing to take under account is that the user ID should be saved to avoid the tenuous chance that another user could obtain the same unique ID in the future and gain control over the server resources that the old user has been granted access to.

**Comment by eldadfux** [comment]
We can add a 'dummy' collection and use it to keep the user ID in our database by updating the internal collection type from users to the 'reserved' collection and erase all other data related to the user.

When generating new ID's we are always verifying no duplicate IDs exist, so once we still have the user ID on the dummy collection, we should be fine. We might be able to use this collection in the future for other sensitive resources we want to allow to delete.

We can also leverage the deletes worker to delete user-related data like sessions and team memberships.

**Comment by TorstenDittmann** [comment]
### Current state

- delete endpoint at `DELETE` /v1/users/:userId
- deletes user
- deletes all sessions
- creates an entry in an internal collection for reserved_uid

### To be done

- check for reserved uid's when creating users
- delete team memberships
- adapt Console UI

The only thing that might cause me difficulties is to delete the memberships to teams. That probably needs some research. But will be done before next sunday 👍

**Comment by eldadfux** [comment]
Great progress @TorstenDittmann ! 

Delete the team memberships should be pretty simple as well. Just fetch all the user membership, iterate them, and delete one-by-one.

The query should look like this:
```php
$memberships = $projectDB->getCollection([
                'limit' => 2000, // TODO add members limit as a config var
                'offset' => 0,
                'filters' => [
                    '$collection='.Database::SYSTEM_COLLECTION_MEMBERSHIPS,
                    'userId='.$userId,
                ],
            ]);
```

**Comment by eldadfux** [comment]
This feature is now available with the release of Appwrite 0.7 ✅
